internal/http/services/owncloud/ocdav/put.go

```diff
@@ -149,6 +151,24 @@ func (s *svc) handlePut(ctx context.Context, w http.ResponseWriter, r *http.Requ
 		return
 	}
 
+	if sRes.Status.Code == rpc.Code_CODE_OK {
+		// the file already exists
+		// will append a timestamp to the end of the file
+		// sorry, this code is not atomic, data loss may occur :'(
+
+		user, ok := ctxpkg.ContextGetUser(ctx)
+		if !ok {
+			log.Error().Msg("error getting user from context")
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+
+		if hasUploaderRole(user) {
+			ref.Path = utils.AppendNowToPath(ref.Path)
+			log.Debug().Msgf("user %s has uploader role, changed file destination to %s", user.Username, ref.Path)
```

Comment:
please never use Msgf with structured loggers. Instead use the proper key value calls
```suggestion
			log.Debug().Str("user", user.Username).Interface("path", ref.Path).Msg("user has uploader role, changed file name")
```